core/embed/rust/src/ui/model_tt/component/loader.rs

```diff
@@ -173,6 +174,15 @@ impl Component for Loader {
                 } else if self.is_completely_shrunk(now) {
                     return Some(LoaderMsg::ShrunkCompletely);
                 } else {
+                    #[cfg(feature = "haptic")]
+                    {
+                        if matches!(self.state, State::Growing(_)) {
```

Comment:
The code is quite clear but using quite a lot of hard-coded constants:

Maybe we should introduce a new constant in the haptic header - `HAPTIC_MAX_AMPLITUDE`, and also use a constant instead of 1000 (if I remember correctly, we have `LOADER_MAX` somewhere), then modify the expression to use a percentage of `HAPTIC_MAX_AMPLITUDE`.

Reply:
Modified the parameter to the play_custom function to accept percentage directly, which at least hide possible differences in max amplitude between different driver ICs. We lose some precision but i find it acceptable, otherwise we might switch to tenths of percents perhaps.

Also named the constants in loader module

 https://github.com/trezor/trezor-firmware/commit/021f2a41b9ad1499cf7f4fd84f6bdf98f7724df9
https://github.com/trezor/trezor-firmware/commit/34bea44a737782d325742fe0c22bbdfc01afc85f